doc/rst/tools/protoc-gen-chpl/protoc-gen-chpl.rst

```diff
@@ -0,0 +1,198 @@
+===============
+Chapel Protobuf
+===============
+
+Chapel Protobuf is a Google Protocol Buffers implementation for Chapel.
+`Protocol Buffers`_ is a language-neutral, platform-neutral, extensible mechanism
+for serializing structured data. The protocol buffer language is a language for 
+specifying the schema for structured data. This schema is compiled into language
+specific bindings.
+
+This project is comprised of two components:
+
+* Code generator: The ``protoc-gen-chpl`` tool is a compiler plugin to ``protoc``, the protocol
+  buffer compiler. It enables the ``protoc`` compiler to generate Chapel code.
+  
+* Chapel library: A module that has the runtime implementation of protobufs
+  in Chapel and provides functionality to serialize messages in wire format.
+
+
+Installation Instructions
+=========================
+
+You should have the protocol buffer compiler installed on your system. For example,
+to install protobuf on an Ubuntu system, you might use ``apt-get install libprotobuf-dev``.
+Also please see the official installation `guide`_.
+
+Once you have the compiler installed, in ``$CHPL_HOME`` run the following:
+
+.. code-block:: sh
+
+   make protoc-gen-chpl
+  
+It builds the protoc-gen-chpl binary so that the command line interface can be used.
+This installs protoc-gen-chpl in the same place as the chapel compiler (``chpl``) so that
+it can be used anywhere in the user's file system.
+
+To remove protobuf suppot, change directory to ``$CHPL_HOME/tools/protoc-gen-chpl`` and run:
+
+.. code-block:: sh
+
+   make clean
+
+
+Defining protocol format
+========================
+
+To make use of the Chapel protobuf implementation you need to start by defining
+a ``.proto`` file. The definitions in a ``.proto`` file contain a message for each
+data structure you want to serialize, then specify a name and a type for each 
+field in the message.
+
+Below is an example of an ``addressbook`` for a person. This section descibes a
+simple ``.proto`` file and the corresponding generated chapel code. For complete
+details on ``.proto`` files see the links at the end of this document.
+
+The Chapel implementation only supports the `proto3`_ syntax of protobuf. The
+``.proto`` file starts with an optional package declaration, which helps to prevent
+naming conflicts between different projects.
+
+.. code-block:: proto
+
+  syntax = "proto3";
+  package addressbook;
+
+In Chapel the generated records will be placed in a module matching the ``package``
+name. If the ``package`` name is not specified the module takes the name of the
+proto file with all non-aplhanumeric characters replaced by an ``underscore``.
+
+Next, you have to define your messages. A message is just an aggregate containing
+a set of typed fields. Many standard simple data types are available as field types,
+including int64, int32, float, double, and string.
```

Comment:
Would it be helpful to call out the Chapel types these translate to here?

Reply:
I guess, it would be better to have it in the detailed guide for the generated code.

What do you say?

Reply:
Seems reasonable to me :)